### PR TITLE
Refactor changelog

### PR DESCRIPTION
**Pull request by ulysses-you**
### _Why are the changes needed?_

refactor the changelog script to fix:
- the nit of Apache Kyuubi(Incubating)
- the format of the output changelog

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: docs/changelog/v1.5.2-incubating.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+## Changelog for Apache Incubating Kyuubi v1.5.2-incubating
```
Comment (yaooqinn) [inline docs/changelog/v1.5.2-incubating.md:1]: it's Apache Kyuubi (Incubating)...

File: docs/changelog/v1.5.2-incubating.md [hunk 2]
```diff
@@ -1,4 +1,5 @@
 ## Changelog for Apache Kyuubi(Incubating) v1.5.2-incubating
+- [Fix changelog nit](https://github.com/apache/incubator-kyuubi/commit/62b51aa3)
```
Comment (yaooqinn) [inline docs/changelog/v1.5.2-incubating.md:2]: you can not do this

File: docs/changelog/v1.5.2-incubating.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
+## Changelog for Apache Kyuubi(Incubating) v1.5.2-incubating
+- [Fix changelog nit](https://github.com/apache/incubator-kyuubi/commit/62b51aa3)
+- [[KYUUBI #2746][INFRA][1.5] Improve NOTICE of binary release](https://github.com/apache/incubator-kyuubi/commit/35a4c488)  
+- [[KYUUBI-2422] Wrap close session with try-finally (#2836)](https://github.com/apache/incubator-kyuubi/commit/cbca761a)
```
Comment (yaooqinn) [inline docs/changelog/v1.5.2-incubating.md:4]: this line crashes too

File: build/release/generate_changelog.sh [hunk 4]
```diff
@@ -49,6 +49,6 @@ if [ "${LAST_RELEASE_TAG}" = "-h" ]; then
 fi
 
 ## Add title for changelog doc.
-echo "## Changelog for Apache Incubating Kyuubi $RELEASE_VERSION  " > $CHANGELOG_PATH
+echo "## Changelog for Apache Kyuubi(Incubating) $RELEASE_VERSION  " > $CHANGELOG_PATH
 ## Append well-formatted git log to changelog file.
 git log --pretty="- [%s]($KYUUBI_GITHUB_COMMIT_URL%h)  " $LAST_RELEASE_TAG..$CURRENT_RELEASE_TAG | grep -v "\[RELEASE\]" >> $CHANGELOG_PATH
```
Comment (yaooqinn) [inline build/release/generate_changelog.sh:54]: let's remove the list-item marker `-`

**Comment by ulysses-you** [comment]
@yaooqinn refactor the changelog

**Comment by yaooqinn** [comment]
you can break this PR into
 1) fix the changelog script + 1.5.1
 2) 1.5.2

File: build/release/generate_changelog.sh [hunk 5]
```diff
@@ -49,6 +49,6 @@ if [ "${LAST_RELEASE_TAG}" = "-h" ]; then
 fi
 
 ## Add title for changelog doc.
-echo "## Changelog for Apache Incubating Kyuubi $RELEASE_VERSION  " > $CHANGELOG_PATH
+echo "## Changelog for Apache Kyuubi(Incubating) $RELEASE_VERSION" > $CHANGELOG_PATH
 ## Append well-formatted git log to changelog file.
-git log --pretty="- [%s]($KYUUBI_GITHUB_COMMIT_URL%h)  " $LAST_RELEASE_TAG..$CURRENT_RELEASE_TAG | grep -v "\[RELEASE\]" >> $CHANGELOG_PATH
\ No newline at end of file
+git log --pretty="[%s]($KYUUBI_GITHUB_COMMIT_URL%h)%n" $LAST_RELEASE_TAG..$CURRENT_RELEASE_TAG | grep -v "\[RELEASE\]" >> $CHANGELOG_PATH
```
Comment (yaooqinn) [inline build/release/generate_changelog.sh:54]: %n is also unnecessary

**Comment by ulysses-you** [comment]
> you can break this PR into
> 
> 1. fix the changelog script + 1.5.1
> 2. 1.5.2

@yaooqinn reverted the 1.5.2 changelog

File: docs/changelog/v1.5.1-incubating.md [hunk 6]
```diff
@@ -1,10 +1,10 @@
...
-- [[KYUUBI #2266] The default value of frontend.connection.url.use.hostname should be set to true to be consistent with previous versions](https://github.com/apache/incubator-kyuubi/commit/d3e25f0)  
-- [[KYUUBI #2255]The engine state of Spark's EngineEvent is hardcoded with 0](https://github.com/apache/incubator-kyuubi/commit/2af8bbb)  
-- [[KYUUBI #2008][FOLLOWUP] Support engine type and subdomain in kyuubi-ctl](https://github.com/apache/incubator-kyuubi/commit/d1a2dda)  
-- [[KYUUBI #2156][FOLLOWUP] Fix configuration format in document](https://github.com/apache/incubator-kyuubi/commit/5225b54)  
-- [[KYUUBI #2156] Change log to reflect exactly why getting token failed](https://github.com/apache/incubator-kyuubi/commit/21ca754)  
-- [[KYUUBI #2134] Respect Spark bundled log4j in extension modules](https://github.com/apache/incubator-kyuubi/commit/30dc84b)  
-- [[KYUUBI #2150] [DOCS] Fix Getting Started With Kyuubi on Kubernetes](https://github.com/apache/incubator-kyuubi/commit/e232a83)  
+## Changelog for Apache Kyuubi(Incubating) v1.5.1-incubating
+[[KYUUBI #2354] Fix NPE in process builder log capture thread](https://github.com/apache/incubator-kyuubi/commit/5e76334e)
```
Comment (yaooqinn) [inline docs/changelog/v1.5.1-incubating.md:2]: you should not remove 2 tailing spaces

**Comment by yaooqinn** [comment]
LGTM except the pr desc

**Comment by ulysses-you** [comment]
@yaooqinn updated the description

**Comment by ulysses-you** [comment]
thanks, merging to master

**Pull request closed without merging**